src/backend/starknet.cairo

```diff
@@ -295,4 +302,53 @@ namespace Internals {
 
         return _save_storage(starknet_address, storage_start + DictAccess.SIZE, storage_end);
     }
+
+    func _save_valid_jumpdests{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        starknet_address: felt, dict_start: DictAccess*, dict_end: DictAccess*
+    ) {
+        alloc_locals;
+        let (local keys_start: felt*) = alloc();
+        let dict_len = dict_end - dict_start;
+        let (local len, _) = unsigned_div_rem(dict_len, DictAccess.SIZE);
```

Comment:
no need to use a range_check, there is no way to have `dict_len` not a multiple of `DictAccess.SIZE`.

But in any case you don't really need this, you can retrieve in the body loop `dict_end` with `[fp - 3]`, update `dict_start = dict_start + DictAccess.SIZE` and break when start == end

Reply:
*to be compared in step with having this extra `len` variable in the loop, looks like it the same for the loop part, but you'd still save the initial div